baseplate/observers/tracing.py

```diff
@@ -179,6 +180,12 @@ def __init__(self, service_name, hostname, span, recorder):
         self.elapsed = None
         self.binary_annotations = []
         self.on_set_tag(ANNOTATIONS["COMPONENT"], "baseplate")
+
+        # Set a debug annotation for downstream processing to
+        # utilize in span filtering.
+        if self.span.flags and (self.span.flags & FLAGS["DEBUG"]):
```

Comment:
should we do this in `finish()` so the user can do `request.trace.flags = DEBUG` mid-request to turn it on?